### ISSUE TITLE
[Feature] Change Web UI port

### ISSUE THREAD

**Issue by gantoine** [rommapp/romm#611]
<div type='discussions-op-text'>

<sup>Originally posted by **Budlyte** January 17, 2024</sup>
Is it possible in the config to change from port 8080, on the container side?  If routing through a VPN container, qbittorrent already uses 8080.

Or could the main build be changed to use a different port by default?</div>

@Budlyte: I'm using binhex-qbittorrentvpn, looks like it works the same way as gluetun, but we're stuck using the port of the container.

**Comment by gantoine** [comment]
Not possible at the moment, and likely not an option we're going to add any time soon (due to technical challenges).

@zurdi15 we could always change the port to something random in the future.

**Comment by bricas** [comment]
As a way forward, can you just change the qbittorrent port with the `WEBUI_PORT` env var?

**Comment by Budlyte** [comment]
> As a way forward, can you just change the qbittorrent port with the `WEBUI_PORT` env var?

Maybe it's just my instance, but the app doesn't like that and restarts, putting 8080 back in the config.

**Comment by Casuallynoted** [comment]
> > As a way forward, can you just change the qbittorrent port with the `WEBUI_PORT` env var?
> 
> Maybe it's just my instance, but the app doesn't like that and restarts, putting 8080 back in the config.

That's really odd, what platform are you currently running the Qbittorrent container on? I'm currently using Docker with kubernetes and have been able to change it via env and it's taken.

**Comment by Budlyte** [comment]
Unraid

Tried just changing it directly in the config file and the option in the UI, which also updates the config file. Still it just locks up, then restarts with 8080

I think just using a port that isn't already in use by a popular application would be best anyway.

**Comment by Jinorex** [comment]
@Budlyte if your using unraid there shouldn't be a problem with changing the port.
I tried changing 8080 to 8085 and it works fine. But i'm not using the available Unraid Community App. 

Quick look at the unraid app and the default Port is 8091 and not 8080. So changing the port shouldn't be a problem.
![image](https://github.com/zurdi15/romm/assets/35707423/2e1069c1-c5d5-45e6-927a-25e565c7c266)

**Comment by Eurotimmy** [comment]
Hey @Budlyte, just confirming I use 8091 successfully in the Unraid CA template.

**Comment by Budlyte** [comment]
Yeah, guys you're correct, you can change the host port without issue. This is basic container management and has nothing to do with the web server port (8080) that RomM itself is built to use.

Please re-read the original post - this issue is with routing Romm through another container that hosts a VPN. If already using QB (which also uses port 8080), a very popular bittorrent application, then the user has to choose one or the other.

If messing around in Unraid, try a configuration like what I did with Deemix here. It's network access is through my Binhex-QB container, which hosts my VPN. And so Deemix, on port 6595, is accessed through that QB/VPN container. I could change the host side port all I want, but port 6595 still has to exist on the container side.

![image](https://github.com/zurdi15/romm/assets/4451365/509c5051-fed7-4c3b-aa80-f108639f2f00)

**Comment by lsaranto** [comment]
I'm running RomM on a dedicated IP. Having the port on 8080 is a bit pointless. Is it not possible to make the port configurable by an environment variable?

**Comment by Budlyte** [comment]
> I'm running RomM on a dedicated IP. Having the port on 8080 is a bit pointless. Is it not possible to make the port configurable by an environment variable?

You're not rubbing in docker?

**Comment by lsaranto** [comment]
Yes, it's in docker. But it has a dedicated IP for easy access and (later) secure guest access.

**Comment by adamantike** [comment]
The next release will allow setting the ROMM_PORT environment variable to customize the port to a value different than 8080.

**Comment by drbloop** [comment]
Note that with this change if you run romm in Kubernetes with Deployment name "romm" you need to explicitly set environment variable `ROMM_PORT: "8080"` or the app will pick up the existing `ROMM_PORT` environment variable that Kubernetes generates and the app refuses to start. Ran into this upgrading to 3.8.0.
